perf/CMakeLists.txt

```diff
@@ -11,7 +24,7 @@ if (NOT ${benchmark_FOUND})
                       COMMENT ${MSG}
     )
     add_custom_target(test-perf
-                      DEPENDS test-c-perf test-lua-perf
```

Comment:
I propose to make independent targets for running tests and aggregating results. Discussed verbally and PoC is below:

```diff
diff --git a/perf/CMakeLists.txt b/perf/CMakeLists.txt
index 805a8eaf78..609ebfe608 100644
--- a/perf/CMakeLists.txt
+++ b/perf/CMakeLists.txt
@@ -5,9 +5,11 @@ file(MAKE_DIRECTORY ${PERF_OUTPUT_DIR})
 
 add_subdirectory(lua)
 
+set(BENCH_RESULTS "")
+
 set(TARANTOOL_BIN $<TARGET_FILE:tarantool>)
 add_custom_target(test-perf-aggregate
-                  DEPENDS test-c-perf test-lua-perf
+                  DEPENDS ${BENCH_RESULTS}
                   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
                   COMMAND ${TARANTOOL_BIN} ${CMAKE_CURRENT_SOURCE_DIR}/tools/aggregate.lua
                     --output=${PERF_OUTPUT_DIR}/summary.txt
@@ -24,7 +26,7 @@ if (NOT ${benchmark_FOUND})
                       COMMENT ${MSG}
     )
     add_custom_target(test-perf
-                      DEPENDS test-perf-aggregate
+                      DEPENDS test-c-perf test-lua-perf
                       COMMENT "Running performance tests"
     )
     return()
@@ -51,14 +53,17 @@ function(create_perf_test_target)
                         "${multiValues}"
                         ${ARGN})
   message(STATUS "Creating C performance test ${PERF_TARGET}_perftest")
+  set(BENCH_RESULT ${PERF_OUTPUT_DIR}/${PERF_TARGET}.json)
   add_custom_target(${PERF_TARGET}_perftest
                     COMMAND "$<TARGET_FILE:${PERF_TARGET}.perftest>"
                             "--benchmark_out_format=json"
-                            "--benchmark_out=${PERF_OUTPUT_DIR}/${PERF_TARGET}.json"
+                            "--benchmark_out=${BENCH_RESULT}"
                     DEPENDS ${PERF_TARGET}.perftest
+                    BYPRODUCTS ${BENCH_RESULT}
                     COMMENT Running ${PERF_TARGET}_perftest
   )
   set(RUN_PERF_C_TESTS_LIST ${RUN_PERF_C_TESTS_LIST} ${PERF_TARGET}_perftest PARENT_SCOPE)
+  set(BENCH_RESULTS "${BENCH_RESULTS} ${BENCH_RESULT}" PARENT_SCOPE)
 endfunction()
 
 function(create_perf_test)
@@ -110,6 +115,6 @@ add_custom_target(test-c-perf
 )
 
 add_custom_target(test-perf
-                  DEPENDS test-perf-aggregate
+                  DEPENDS test-c-perf test-lua-perf
                   COMMENT "Running performance tests"
 )
diff --git a/perf/lua/CMakeLists.txt b/perf/lua/CMakeLists.txt
index 55d523f92e..40000e5ddf 100644
--- a/perf/lua/CMakeLists.txt
+++ b/perf/lua/CMakeLists.txt
@@ -19,17 +19,20 @@ function(create_perf_lua_test)
 
   message(STATUS "Creating Lua performance test ${PERF_NAME}_perftest")
   set(TEST_PATH ${CMAKE_CURRENT_SOURCE_DIR}/${PERF_NAME}.lua)
+  set(BENCH_RESULT "${PERF_OUTPUT_DIR}/${PERF_NAME}.json")
   add_custom_target(${PERF_NAME}_perftest
                     COMMAND ${CMAKE_COMMAND} -E env
                       LUA_PATH="${LUA_PATH}"
                       ${TARANTOOL_BIN} ${TEST_PATH}
-                        --output="${PERF_OUTPUT_DIR}/${PERF_NAME}.json"
+                        --output="${BENCH_RESULT}"
                         --output_format=json
                     COMMENT Running ${PERF_NAME}_perftest
                     DEPENDS tarantool ${TEST_PATH}
                     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+                    BYPRODUCTS ${BENCH_RESULT}
   )
   set(RUN_PERF_LUA_TESTS_LIST ${RUN_PERF_LUA_TESTS_LIST} ${PERF_NAME}_perftest PARENT_SCOPE)
+  set(BENCH_RESULTS "${BENCH_RESULTS} ${BENCH_RESULT}" PARENT_SCOPE)
 endfunction()
 
 create_perf_lua_test(NAME 1mops_write)

```

Reply:
I've introduced two separated targets -- one depends for the results, and the second runs the command unconditionally.

Reply:
Thanks!